code/modules/mob/mob.dm

```diff
@@ -1019,4 +1019,17 @@ var/list/slot_equipment_priority = list( \
 
 //can the mob be unbuckled from something by default?
 /mob/proc/can_unbuckle()
-	return 1
\ No newline at end of file
+	return 1
+
+//Can the mob see reagents inside of containers?
+/mob/proc/can_see_reagents()
+	if(stat == DEAD) //Ghosts and such can always see reagents
+		return 1
+	if(issilicon(src)) //Silicons can automatically view reagents
+		return 1
+	if(ishuman(src))
+		var/mob/living/carbon/human/H = src
+		for(var/obj/item/clothing/C in H) //If they have some clothing equipped that lets them see reagents, they can see reagents
```

Comment:
if you want to loop through actual clothing, check out changeling transformations. i have a list of all clothing slot variables there, so you can do
```
for(var/slotname in slot_list)
    var/obj/item/clothing/C = H.vars[slotname]
```